src/NServiceBus.Core/Pipeline/Outgoing/AttachSenderRelatedInfoOnMessageBehavior.cs

```diff
@@ -2,13 +2,15 @@ namespace NServiceBus
 {
     using System;
     using System.Threading.Tasks;
+    using NServiceBus.Transport;
     using Pipeline;
 
     class AttachSenderRelatedInfoOnMessageBehavior : IBehavior<IRoutingContext, IRoutingContext>
     {
         public Task Invoke(IRoutingContext context, Func<IRoutingContext, Task> next)
         {
             var message = context.Message;
+            var utcNow = DateTimeOffset.UtcNow;
 
             if (!message.Headers.ContainsKey(Headers.NServiceBusVersion))
```

Comment:
Let's add a comment here to document what we learned about auditing.
```suggestion
            // This behavior executes in the case of auditing as well, so assuming there are no dispatch properties set,
            // no existing header should be overwritten, otherwise the message being audited would be modified improperly.

            if (!message.Headers.ContainsKey(Headers.NServiceBusVersion))
```

Reply:
Comment added